src/Servers/Kestrel/Core/src/ListenOptions.cs

```diff
@@ -140,7 +140,8 @@ internal string Scheme
     }
 
     internal bool IsTls { get; set; }
-    internal HttpsConnectionAdapterOptions? HttpsOptions { get; set; }
+    /// <remarks>Should not be inspected until the configuration has been loaded.</remarks>
+    internal Lazy<HttpsConnectionAdapterOptions>? HttpsOptions { get; set; }
```

Comment:
I should have caught this last time, but can we make this a func? This should be called again if the endpoint is rebound due to a default certificate change in config.

I think our best bet is keeping the old `internal HttpsConnectionAdapterOptions? HttpsOptions { get; set; }` property, but then also adding a `Func<HttpsConnectionAdapterOptions>` property called `CreateHttpsConnectionOptions` or something like that that we call in `KestrelServerImple.StartAsync+OnBind` and then set `HttpsOptions`.

Reply:
I'm not sure I understand the point about keeping the old property.  I think you're saying that consumers should be able to specify either an `HttpsConnectionAdapterOptions` or a `Func<HttpsConnectionAdapterOptions>`, which would make it unnecessary to wrap an already-in-hand object in a func, but that seems like an unimportant win in code that isn't perf sensitive and has the downside of requiring an understanding of which takes precedence if both are set.  But maybe you had some other reason in mind?

Reply:
I'm having trouble repro'ing a case where Func vs Lazy causes a problem - can you give me an example?  So far, when I try, the lazy gets clobbered with a fresh value here
```
 	Microsoft.AspNetCore.Server.Kestrel.Core.dll!Microsoft.AspNetCore.Hosting.ListenOptionsHttpsExtensions.UseHttps(Microsoft.AspNetCore.Server.Kestrel.Core.ListenOptions listenOptions, System.Lazy<Microsoft.AspNetCore.Server.Kestrel.Https.HttpsConnectionAdapterOptions> lazyHttpsOptions) Line 213	C#
 	Microsoft.AspNetCore.Server.Kestrel.Core.dll!Microsoft.AspNetCore.Hosting.ListenOptionsHttpsExtensions.UseHttps(Microsoft.AspNetCore.Server.Kestrel.Core.ListenOptions listenOptions, Microsoft.AspNetCore.Server.Kestrel.Https.HttpsConnectionAdapterOptions httpsOptions) Line 196	C#
>	Microsoft.AspNetCore.Server.Kestrel.Core.dll!Microsoft.AspNetCore.Server.Kestrel.KestrelConfigurationLoader.Reload() Line 373	C#
 	Microsoft.AspNetCore.Server.Kestrel.Core.dll!Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerImpl.RebindAsync() Line 352	C#
 	Microsoft.AspNetCore.Server.Kestrel.Core.dll!Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerImpl.TriggerRebind(object state) Line 330	C#
```

Reply:
@halter73 I've resolved everything but this thread.   Can you please elaborate a bit on your concerns?

Reply:
Thinking about this some more, won't a `Lazy<HttpsConnectionAdapterOptions>` be clobbered on rebind, the same way a `HttpsConnectionAdapterOptions` was before my change?

Reply:
I think your right. I forgot that `KestrelConfigurationLoader.Reload` recreates both the ListenOptions and HttpsConnectionAdapterOptions from scratch. This looks good!